ehrapy/preprocessing/_encoding.py

```diff
@@ -85,29 +78,42 @@ def encode(
         if "original" not in adata.layers.keys():
             adata.layers["original"] = adata.X.copy()
 
-        # autodetect categorical values, which could lead to more categoricals
+        # autodetect categorical values based on feature types stored in adata.var[FEATURE_TYPE_KEY]
```

Comment:
in Line 71, instead of if isinstance(adata, AnnData): and then indenting the entire function, it'd be easier to keep track of the indentings when doing 

```py
if not isinstance(adata, AnnData):
    raise ValueError(f"Cannot encode object of type {type(adata)}. Can only encode AnnData objects!")
```

instead, then almost the entire function can be unindented.

Has been like this before, but might as well improve this here :)

Reply:
I changed that - thanks!